eeg_flow/tasks/epochs_evoked.py

```diff
@@ -0,0 +1,434 @@
+# ########################################
+# Modified on Mon May 08 01:01:00 2023
+# @anguyen
+
+from collections import Counter
+import math
+import itertools
+import time
+
+import numpy as np
+import pandas as pd
+from autoreject import get_rejection_threshold
+from mne import Epochs, find_events
+from mne.epochs import make_metadata
+from mne.io import read_raw_fif
+from scipy.stats import norm
+
+from ..config import load_config
+from ..utils._docs import fill_doc
+from ..utils.bids import get_fname, get_derivative_folder
+from ..utils.concurrency import lock_files
+
+
+@fill_doc
+def behav_prep_epoching(
+    participant: str,
+    group: str,
+    task: str,
+    run: int,
+    *,
+    timeout: float = 10,
+) -> None:
+    """Compute behav report and output metadata.
+    Parameters
+    ----------
+    %(participant)s
+    %(group)s
+    %(task)s
+    %(run)s
+    """
+    # prepare folders
+    _, derivatives_folder_root, _ = load_config()
+    fname_stem = get_fname(participant, group, task, run)
+    derivatives_folder = get_derivative_folder(
+        derivatives_folder_root, participant, group, task, run
+    )
+
+    # lock the output derivative files
+    # create locks
+    derivatives = [
+        derivatives_folder / (fname_stem + "_step9_a-metadata.csv"),
+        derivatives_folder / (fname_stem + "_step9_b-behav.txt"),
+        derivatives_folder / (fname_stem + "_step9_c1-cleaned-epo.fif"),
+        derivatives_folder / (fname_stem + "_step9_c2-drop-epochs.csv"),
+        derivatives_folder / (fname_stem + "_step9_c3-drop-channel-log.csv"),
+        derivatives_folder / (fname_stem + "_step9_d1-standard_evoked-ave.fif"),
+        derivatives_folder / (fname_stem + "_step9_d2-target_evoked-ave.fif"),
+        derivatives_folder / (fname_stem + "_step9_d3-novel_evoked-ave.fif"),
+    ]
+
+    locks = lock_files(*derivatives, timeout=timeout)
+
+    try:
+        _behav_prep_epoching(participant, group, task, run)
+    finally:
+        for lock in locks:
+            lock.release()
+        del locks
+    return
```

Comment:
No need to `return None` or `return` explicitly, if nothing is returned at the end of a function it returns `None` anyway. Thus, those 3 functions are equivalent:

```
def foo():
    print ("hello")
    return None

def foo():
    print ("hello")
    return

def foo():
    print ("hello")
```

In all 3 cases, `test = foo()` will set the variable `test` to `None`.
By convention, the `return is omitted, thus the shorter syntax is preferred.